OrbitCore/ConnectionManager.cpp

```diff
@@ -205,53 +209,37 @@ void ConnectionManager::SetupClientCallbacks() {
     GCoreApp->RefreshCaptureView();
   });
 
-  GTcpClient->AddCallback(Msg_SamplingCallstack, [=](const Message& a_Msg) {
-    // TODO: Send buffered callstacks.
-    LinuxCallstackEvent data;
-
-    std::istringstream buffer(a_Msg.GetDataAsString());
-    cereal::JSONInputArchive inputAr(buffer);
-    inputAr(data);
-
-    GCoreApp->ProcessSamplingCallStack(data);
-  });
-
-  GTcpClient->AddCallback(Msg_RemoteTimers, [=](const Message& a_Msg) {
+  GTcpClient->AddCallback(Msg_Timers, [=](const Message& a_Msg) {
     uint32_t numTimers = a_Msg.m_Size / sizeof(Timer);
     const Timer* timers = static_cast<const Timer*>(a_Msg.GetData());
     for (uint32_t i = 0; i < numTimers; ++i) {
       GTimerManager->Add(timers[i]);
     }
   });
 
-  GTcpClient->AddCallback(Msg_KeyAndString, [=](const Message& a_Msg) {
-    KeyAndString key_and_string;
+  GTcpClient->AddCallback(Msg_KeysAndStrings, [=](const Message& a_Msg) {
     std::istringstream buffer(a_Msg.GetDataAsString());
     cereal::BinaryInputArchive inputAr(buffer);
-    inputAr(key_and_string);
-    GCoreApp->AddKeyAndString(key_and_string.key, key_and_string.str);
+    std::vector<KeyAndString> keys_and_strings;
+    inputAr(keys_and_strings);
+
+    for (const auto& key_and_string : keys_and_strings) {
+      GCoreApp->AddKeyAndString(key_and_string.key, key_and_string.str);
+    }
   });
 
-  GTcpClient->AddCallback(Msg_RemoteCallStack, [=](const Message& a_Msg) {
-    CallStack stack;
+  GTcpClient->AddCallback(Msg_LinuxAddressInfos, [=](const Message& a_Msg) {
     std::istringstream buffer(a_Msg.GetDataAsString());
-    cereal::JSONInputArchive inputAr(buffer);
-    inputAr(stack);
+    cereal::BinaryInputArchive inputAr(buffer);
```

Comment:
You could also use DeserializeObjectBinary() from Serialization.h, but it takes in a void* and a size, we could add a version that takes in a string.

Reply:
Added `void DeserializeObjectBinary(const std::string& data, T& object)` and used in these four `GTcpClient->AddCallback`.